core/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala

```diff
@@ -277,6 +288,34 @@ object VacuumCommand extends VacuumCommandImpl with Serializable {
 
 trait VacuumCommandImpl extends DeltaCommand {
 
+  protected def getFilePath(
+      fileAction: FileAction,
+      fs: FileSystem,
+      reservoirBase: Path,
+      relativizeIgnoreError: Boolean): Seq[String] = {
+    val filePath = stringToPath(fileAction.path)
+    val validFileOpt = if (filePath.isAbsolute) {
+      val maybeRelative =
+        DeltaFileOperations.tryRelativizePath(fs, reservoirBase,
+          filePath, relativizeIgnoreError)
+      if (maybeRelative.isAbsolute) {
+        // This file lives outside the directory of the table
+        None
+      } else {
+        Option(pathToString(maybeRelative))
+      }
+    } else {
+      Option(pathToString(filePath))
+    }
+    validFileOpt.toSeq.flatMap { f =>
+      // paths are relative so provide '/' as the basePath.
```

Comment:
I know this was in the original code, but it's confusing to somebody who doesn't already know how `getAllSubdirs` works. Is there a way for the comment to hint at what was achieved, and why, rather than what was done? (reader can see easily enough that base path is "/").

In most path APIs, "base" refers to a prefix that should eventually be prepended to the path name. But here it refers to a common prefix the path name is expected to already contain.

Reply:
`// get all parent paths that file has, those paths will be used to find the untracked files and dirs` could this remove the confusion?

Reply:
Yeah, something like that would be great